router-tests/integration_test.go

```diff
@@ -365,6 +367,25 @@ func prettifyJSON(t *testing.T, jsonStr string) string {
 	return res.String()
 }
 
+func TestQueryWithLogging(t *testing.T) {
```

Comment:
Please put this in a separate file for more visibility and better test structure.